src/fault/Action.cpp

```diff
@@ -59,67 +60,141 @@ void Action::ensure(Type t)
         throw "wrong action type";
 }
 
-Action::Action() : type_(NOP)
+bool Action::is_action_on_field(void) const
+{
+    return ((type_ == Type::BITFLIP || type_ == Type::MASK) ? true : false);
+}
+
+Action::Action()
+  : type_(Type::NOP)
 {
     etiss::log(etiss::VERBOSE, std::string("etiss::fault::Action::Action() called. "));
 }
 
-Action::Action(const InjectorAddress &inj, const std::string &command) : type_(COMMAND), inj_(inj), command_(command)
+#ifndef NO_ETISS
+Action::Action(etiss::int32 event)
+  : type_(Type::EVENT)
+  , event_(event)
+{
+    etiss::log(etiss::VERBOSE, std::string("etiss::fault::Action::Action(etiss::int32 exception) called. "));
+}
+#endif
+
+Action::Action(const InjectorAddress &inj, const std::string &command)
+  : type_(Type::COMMAND)
+  , inj_(inj)
+  , command_(command)
 {
     etiss::log(etiss::VERBOSE, std::string("etiss::fault::Action::Action(InjectorAddress &=") + inj.getInjectorPath() +
                                    std::string(", command=") + command + std::string(") called. "));
 }
 
 Action::Action(const InjectorAddress &inj, const std::string &field, unsigned bit)
-    : type_(BITFLIP), inj_(inj), field_(field), bit_(bit)
+  : type_(Type::BITFLIP)
+  , inj_(inj)
+  , field_(field)
+  , bit_(bit)
 {
     etiss::log(etiss::VERBOSE, std::string("etiss::fault::Action::Action(InjectorAddress &=") + inj.getInjectorPath() +
                                    std::string(", field=") + field + std::string(", bit=") + std::to_string(bit) +
                                    std::string(") called. "));
 }
 
-Action::Action(const Fault &fault) : type_(INJECTION)
+Action::Action(const InjectorAddress &inj, const std::string &field, MaskOp mask_op, uint64_t mask_value)
+  : type_(Type::MASK)
+  , inj_(inj)
+  , field_(field)
+  , mask_op_(mask_op)
+  , mask_value_(mask_value)
+{
+    etiss::log(etiss::VERBOSE, std::string("etiss::fault::Action::Action(InjectorAddress &=") + inj.getInjectorPath() +
+                                   std::string(", field=") + field + std::string(", mask_op=") + std::string(mask_op_) +
+                                   std::string(", mask_value=") + std::to_string(mask_value) +
+                                   std::string(") called. "));
+}
+
+Action::Action(const Fault &fault)
+  : type_(Type::INJECTION)
 {
     etiss::log(etiss::VERBOSE,
                std::string("etiss::fault::Action::Action(Fault &=") + fault.toString() + std::string(") called. "));
     fault_.push_back(fault);
 }
 
-Action::Type Action::getType() const
+const Action::type_t& Action::getType() const
 {
     return type_;
 }
 
 const InjectorAddress &Action::getInjectorAddress() const
 {
+    if(unlikely(!(type_ == Type::BITFLIP || type_ == Type::MASK || type_ == Type::COMMAND)))
```

Comment:
unless they are measurably important for performance, i'd perfer to avoid the un/likely clutter

Reply:
agreed. From old code base, but the getters are not performance critical. Fixed with https://github.com/tum-ei-eda/etiss/pull/100/commits/b6da677f88d58799f6b97391e25baf7845873e54